### ISSUE TITLE
Failed to build libchromiumcontent for Electron 1.7.9

### ISSUE THREAD

**Issue by BSoD-Ultimate** [electron/electron#11161]
* Electron version: 1.7.9
* Operating system: Windows 8.1

I need to build Electron from source code for convenience debugging my native node modules. After building debug version of Electron 1.7.9, I discovered the shared library version of libchromiumcontent also needs to be rebuilt from source since no PDB files related to prebuilt libcc libraries was found. I found the script `bootstrap.py` provides a switch `--build-libchromiumcontent` can be used to build libcc however only static library version build is supported. 
Due to these reasons, I checked out the `libchromiumcontent` repository then switched to the commit `62a432e7be8d0b1f6ad7181e3a1e9dd44ad9621c` related to Electron release 1.7.9 which lies in branch `electron-1-7-x`. Then installed these components following Chromium build instructions and the readme markdown file:
- Visual Studio 2015 Update 3 (with win10 SDK 10.0.14393.0)
- Python 2.7.14 (with `PATH` environment variable added and pypiwin32 installed)
- Windows Driver Kit 10.0.16299.15 (for missing 'cdb.exe' in script `script/update` )

When executing build script to build libcc from source using command `python script/build -t ia32 -D`, It generated the following error:

```
I:\libchromiumcontent>python script/build -t ia32 -D
ninja: Entering directory `src\out-ia32\shared_library'
[4/19844] CXX obj/base/base_static/pe_image.obj
FAILED: obj/base/base_static/pe_image.obj
ninja -t msvc -e environment.x86 -- "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64_x86/cl.exe" /nologo /showIncludes /FC @obj/base/base_stati
c/pe_image.obj.rsp /c ../../base/win/pe_image.cc /Foobj/base/base_static/pe_imag
e.obj /Fd"obj/base/base_static_cc.pdb"
i:\libchromiumcontent\src\base\win\pe_image.cc(8): fatal error C1083: Cannot open include file: 'stddef.h': No such file or directory
[5/19844] CXX obj/base/base_paths/base_paths_win.obj
FAILED: obj/base/base_paths/base_paths_win.obj
ninja -t msvc -e environment.x86 -- "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64_x86/cl.exe" /nologo /showIncludes /FC @obj/base/base_paths
/base_paths_win.obj.rsp /c ../../base/base_paths_win.cc /Foobj/base/base_paths/b
ase_paths_win.obj /Fd"obj/base/base_paths_cc.pdb"
c:\program files (x86)\windows kits\10\include\10.0.14393.0\um\winnt.h(34): fata
l error C1083: Cannot open include file: 'ctype.h': No such file or directory
[6/19844] CXX obj/base/base_paths/base_paths.obj
FAILED: obj/base/base_paths/base_paths.obj
ninja -t msvc -e environment.x86 -- "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64_x86/cl.exe" /nologo /showIncludes /FC @obj/base/base_paths
/base_paths.obj.rsp /c ../../base/base_paths.cc /Foobj/base/base_paths/base_path
s.obj /Fd"obj/base/base_paths_cc.pdb"
i:\libchromiumcontent\src\base\files\file_path.h(105): fatal error C1083: Cannot open include file: 'stddef.h': No such file or directory
[9/19844] ACTION //build/win:default_e...en_manifest(//build/toolchain/win:x86)
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "script/build", line 61, in <module>
    sys.exit(main())
  File "script/build", line 45, in main
    subprocess.check_call([NINJA, '-C', os.path.relpath(out_dir), target], env=e
nv)
  File "C:\python27\lib\subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['I:\\libchromiumcontent\\vendor\\depot_
tools\\ninja.exe', '-C', 'src\\out-ia32\\shared_library', 'chromiumcontent:chrom
iumcontent']' returned non-zero exit status 1
```

### Expected behavior

The libcc build procedure ran OK without any warnings and errors.

### How to reproduce

At first, I thought I might wrongly configure the installed Visual Studio/Windows SDK on my computer. Then I tried to re-install Visual Studio and Windows SDKs and the re-install operation completed without any errors occurred. However, the build script **still raised the same error** after the re-install operation. I even reset the computer using Windows 8.1's refresh pc feature then re-install all dependencies from the beginning, and the result is the same. Installing a clean Windows 8.1 on a virtual machine then re-configure tools required also got the same result.

I googled using some filenames with error occurred during the compilation and got several same issues related to Chromium. Here are the results I found:
- https://stackoverflow.com/questions/42597166/getting-no-such-file-or-directory-when-i-try-to-build-chromium
- https://groups.google.com/a/chromium.org/forum/#!searchin/chromium-dev/pe_image$20stddef/chromium-dev/IozdcexRCvY/U7tOWcwuAgAJ

The second one told me CRT headers such as `stddef.h` locate in  the path `C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt`. This header file is exactly located there. However, Ninja could not find it due to an unknown reason.

I tried to switch to `master` branch of libcc and then build libcc from source, everything was OK. Building libcc which integrates into Electron during build bootstrap step since 1.8.x is also OK. Both shared library version of libcc and Electron debug binary successfully generated without any errors. 
The reason why I do not use 1.8.x instead is debugging protocol seems to have changed since Electron 1.8.x. The new debugging protocol seems not compatible with existing Visual Studio Code or `chrome://inspect` page. I can't set any breakpoints in entry JS File to step code in Electron 1.8.x using both `--inspect` and `--inspect-brk` command-line switch during startup. The program simply continues the execution after debugger was attached to the inspect port. Here is the issue I reported before: #11121

I am wondering why compiling libcc on the same commit fails again and again on my computer although at least one successful build has been released(the one downloaded when preparing Electron 1.7.9 build environment ).
I would also appreciate if the prebuilt binary of libcc based on the same commit with debug symbols(PDB files) can be provided.

**Comment by BSoD-Ultimate** [comment]
Well, I tried several different ways to solve the problem. I installed Visual Studio 2017 with all available Windows 10 SDKs selected in installation options. Then the build run smoothly without any errors occurred. There must have some issues with Microsoft's Windows SDKs installation scripts. This phenomenon is really weird.
Here are the Windows SDKs I installed during the Visual Studio 2017 installation:

![3 w sr u 6y 6ok 0xh 0 e](https://user-images.githubusercontent.com/11455017/32987003-b2bd14ec-cd19-11e7-876a-260939a78a52.png)
